packages/fxa-content-server/app/scripts/lib/payment-server.js

```diff
@@ -41,6 +41,13 @@ const PaymentServer = {
       ? account.pick('metricsEnabled').metricsEnabled
       : true;
 
+    if (metricsEnabled === false) {
+      // Strip out the flow params from the query params. We don't want to propagate them.
+      delete queryParams.deviceId;
+      delete queryParams.flowBeginTime;
+      delete queryParams.flowId;
+    }
```

Comment:
@vbudhram : I thought the content server was the only place that added these query params. Why do we need to explicitly remove them here?

Is it related to [this comment](https://github.com/mozilla/fxa/pull/15196/files#r1173067139)?

Reply:
Closing the loop here, apparently RPs can also add flow params for "top of funnel" metrics. We add them if `metricsEnabled` is not `false` and if not provided.